app/src/main/java/com/github/se/project/ui/profile/ProfileInfo.kt

```diff
@@ -92,71 +120,181 @@ fun ProfileInfoScreen(
+                                    alpha = 0.5f))) {
+                      Column(
+                          modifier = Modifier.padding(16.dp),
+                          verticalArrangement = Arrangement.spacedBy(16.dp)) {
+                            // Profile Name
+                            Row(
+                                modifier = Modifier.fillMaxWidth().testTag("profileInfoRow"),
+                                horizontalArrangement = Arrangement.spacedBy(16.dp),
+                                verticalAlignment = Alignment.CenterVertically) {
+                                  // Profile Icon Surface
+                                  Surface(
+                                      modifier = Modifier.size(48.dp),
+                                      shape = MaterialTheme.shapes.medium,
+                                      color =
+                                          MaterialTheme.colorScheme.primary.copy(alpha = 0.1f)) {
+                                        Icon(
+                                            imageVector = Icons.Default.Person,
+                                            contentDescription = null,
+                                            modifier = Modifier.padding(8.dp),
+                                            tint = MaterialTheme.colorScheme.primary)
+                                      }
+
+                                  // Profile Name and Academic Info in a Column
+                                  Column(modifier = Modifier.weight(1f)) {
+                                    // Profile Name
+                                    Text(
+                                        text =
+                                            "${userProfile.firstName.capitalizeFirstLetter()} ${userProfile.lastName.capitalizeFirstLetter()}",
+                                        style = MaterialTheme.typography.titleMedium,
+                                        modifier = Modifier.testTag("profileName"),
+                                        color = MaterialTheme.colorScheme.onBackground)
+
+                                    // Section and Academic Level
+                                    Text(
+                                        text =
+                                            "${userProfile.section} - ${userProfile.academicLevel}",
+                                        style = MaterialTheme.typography.bodyMedium,
+                                        color = MaterialTheme.colorScheme.onSurfaceVariant,
+                                        modifier = Modifier.testTag("profileAcademicInfo"))
+                                  }
+                                }
+
+                            // draw a line between the profile name and the rest of the profile info
+                            Divider(color = MaterialTheme.colorScheme.outlineVariant)
+
+                            // Role Row
```

Comment:
Separating this screen into different rows is a nice adjustment, both for UI and code clarity